### PR TITLE
SDK-980 Dynamic Scenarios - optional attributes

### PR DESCRIPTION
**Pull request by echarrod**
Since it wasn't possible to make everything private which used the optional functionality, I've removed the optional parts for now. We can revert this change when optional attributes are enabled in the app.

### CODE REVIEW

File: src/Yoti.Auth/ShareUrl/Policy/DynamicPolicy.cs [hunk 1]
```diff
@@ -17,19 +17,14 @@ public class DynamicPolicy
         [JsonProperty(PropertyName = "wanted_remember_me")]
         private readonly bool _wantedRememberMeId;
 
-        [JsonProperty(PropertyName = "wanted_remember_me_optional")]
```
Comment (davidgrayston) [inline src/Yoti.Auth/ShareUrl/Policy/DynamicPolicy.cs]: As you already mentioned to me, we may want to keep this property and set to `false`

File: test/Yoti.Auth.Tests/TestData/DynamicPolicy.json [hunk 2]
```diff
@@ -5,27 +5,26 @@
...
       {
         "name": "date_of_birth",
         "derivation": "age_under:40",
-        "optional": true
+        "optional": false
       }
     ],
     "wanted_auth_types": [ 2 ],
-    "wanted_remember_me": false,
-    "wanted_remember_me_optional": false
```
Comment (davidgrayston) [inline test/Yoti.Auth.Tests/TestData/DynamicPolicy.json]: We'd need to reinstate the `wanted_remember_me_optional` property here if you set to `false`

Comment (echarrod) [reply test/Yoti.Auth.Tests/TestData/DynamicPolicy.json]: Yep, updated in 179b5be

File: src/Yoti.Auth/ShareUrl/Policy/DynamicPolicy.cs (hunk 1 above)
Comment (echarrod) [reply src/Yoti.Auth/ShareUrl/Policy/DynamicPolicy.cs]: I've now done that, completed in 179b5be

**Pull request merged**